edx/edx-ora2
# Explicitly Set File Upload Extensions

**TL;DR -** to enable better file upload experiences, we need to better establish and communicate what files learners are allowed to upload for responses. We don't have a list of allowed files types for our existing presets (image, and pdf and image).

**What changed?**
This PR establishes an "allowed extension" list (sourced from the [MDN MIME types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types) page) for the 2 presets and uses it to apply default `white_listed_file_types` values where otherwise unset for the 2 touch points:
1. Loading the `white_listed_file_types` in submission context (student and instructor preview views).
  Behavior is modified to *always* pass the allowed file types list, instead of only for the "custom" file upload configuration.
2. Loading the `white_listed_file_types` string when editing ORA settings (ORA settings view)

**Why?**
This enables:
- Previewing the allowed extensions when configuring an ORA ([EDUCATOR-5344](https://openedx.atlassian.net/browse/EDUCATOR-5344))
- Previewing the allowed extensions when submitting a file response ([EDUCATOR-5345](https://openedx.atlassian.net/browse/EDUCATOR-5345))

Also added/updated some tests to reflect new functionality.

**How should I test this?**
Check the passed context at the following points for ORAs w/ file uploads enabled:
- LMS: `openassessment.xblock.submission_mixin.py::submission_path_and_context()`
- Studio: `openassessment.xblock.studio_mixin.py::editor_context()`

Verify context:
- ORA w/ `image` or `pdf-and-image` file upload type have extensions for `white_listed_file_types`
- ORA w/ `custom` file upload type has configured extensions for `white_listed_file_types`

Alternatively, surface the `white_listed_file_types` variable in Submission/Studio templates.

**Developer Checklist**

JIRA: [EDUCATOR-5346](https://openedx.atlassian.net/browse/EDUCATOR-5346)

FIY: @edx/masters-devs-gta
